libsolidity/formal/Predicate.h

```diff
@@ -166,6 +168,12 @@ class Predicate
 	/// nullptr if this predicate is not associated with a specific program AST node.
 	ASTNode const* m_node = nullptr;
 
+	/// The ContractDefinition that contains this predicate.
+	/// nullptr if this predicate is not associated with a specific contract.
+	/// This is unfortunately necessary because of virtual resolution for
+	/// function nodes.
+	ContractDefinition const* m_contractContext = nullptr;
```

Comment:
Can this be `nullptr` currently?

Reply:
`createSymbolicBlock()` has `nullptr` as a default value for it and I guess there might be calls not visible in the diff that do not set its value explicitly.

Reply:
I don't think so, but it will be possible when we have free functions.

Reply:
Currently they all set it explicitly, but `m_currentContract` might be nullptr when we have free functions.